thinc/backends/_custom_kernels.py

```diff
@@ -74,26 +134,40 @@ def clipped_linear(
     threads_per_block=128,
     num_blocks=128,
 ):
-    _check_compatible(X)
+    _is_float_array(X)
 
     out = X
     if not inplace:
-        out = cupy.zeros_like(X, dtype="f")
-    clipped_linear_kernel(
-        (num_blocks,),
-        (threads_per_block,),
-        (out, X, slope, offset, min_val, max_val, X.size),
-    )
+        out = cupy.zeros_like(X)
+    if X.dtype == "float32":
+        clipped_linear_kernel_float(
+            (num_blocks,),
+            (threads_per_block,),
+            (out, X, slope, offset, min_val, max_val, X.size),
+        )
+    else:
+        clipped_linear_kernel_double(
```

Comment:
Should we add an additional `assert` here on the `dtype`? Or is that not necessary because we've already done the `_is_float_array` check?

Reply:
Yeah, the idea is that the invariants are already checked at the beginning of each function, so we can rely on `dtype` to be `float64` in that case.